shadowtools/src/shadowtools/shadow_exec.py

```diff
@@ -111,11 +113,20 @@ def _main(
     config_path.write_text(yaml.safe_dump(config))
 
     data_dir = tmpdir.joinpath("shadow.data")
-    shadow_args = [
-        f"--data-directory={data_dir}",
-        "--progress=false",
-        str(config_path),
-    ]
+    if any((re.match(r"^--data-directory[=$]|^-d", s) for s in shadow_args)):
+        # It wouldn't be *terribly* hard to support this, but not today.
+        # Naively allowing this override would break our stdout pass-through
+        # below.
+        print(
+            f"ERROR: Overriding shadow's --data-directory currently unsupported.",
+            file=stderr,
+        )
+        sys.exit(1)
+    else:
+        shadow_args.append(f"--data-directory={data_dir}")
+    if not any((re.match(r"^--progress[=$]", s) for s in shadow_args)):
+        shadow_args.append(f"--progress=false")
```

Comment:
Does it make sense to put `progress` in the generated config file instead so that we don't need extra handling here?